File: length_of_last_word.cpp
#include "header.h"

int lengthOfLastWord(const char *s) {
    if(strlen(s)==0) return 0;
    const char* curr = s;
    const char* pre = s;
    int res = 0;
    while(*curr){
        if(*curr!=' '){
            res = curr-pre+1;
            curr++;
        }
        else{
            while(*curr && *curr == ' ')
                curr++;
            if(!*curr) break;
            pre = curr;
            res = curr-pre+1;
        }
    }
    return res;
}

int main(){
    char s[] = "Hello World s ";
    printf("%d\n", lengthOfLastWord(s));
    return 0;
}

File: length_of_last_word.py
class Solution:
    # @param s, a string
    # @return an integer
    def lengthOfLastWord(self, s):
        seen_space = False
        start = 0
        curr = 0
        res = 0
        while curr<len(s):
            if s[curr] != ' ':
                if seen_space:
                    start = curr
                    seen_space = False
                curr += 1
                res = curr-start
                continue
            if not seen_space:
                seen_space = True
            curr += 1
        return res

print Solution().lengthOfLastWord("HELLO WORLD  ")
